subsys/bluetooth/services/fast_pair/include/fp_battery.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _FP_BATTERY_H_
+#define _FP_BATTERY_H_
+
+#include <bluetooth/services/fast_pair.h>
+
+/**
+ * @defgroup fp_battery Fast Pair battery
+ * @brief Internal API for Fast Pair battery module implementation
+ *
+ * Fast Pair battery module provides API for application to set battery data and for Fast Pair
+ * subsystem to get battery data and use it when generating advertising data.
+ *
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Get battery data.
+ *
+ * @return Battery data.
+ */
+struct bt_fast_pair_battery_data fp_battery_get_battery_data(void);
```

Comment:
In that case I would return struct (as it is now) to prevent providing pointer to module's internal data.